src/mygrad/io.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import typing
+import mygrad as mg
+from mygrad.tensor_base import tensor, Tensor
+
+
+def save(filename : str, tensor : Tensor) -> None:
+    """Saves a tensor and its gradient information.
+
+    This docstring was adapted from that of numpy.save()
+
+    Parameters
+    ----------
+    file_name : str
+        The desired name of the file that will hold the tensor data. Note that the file will be saved as a .npz
+
+    tensor : Tensor
+        The tensor that is to be saved, along with its gradient information. If there is no gradient, it saves as None.
+
+    Returns
+    -------
+    None
+    """
+    if not isinstance(tensor, Tensor):
+        raise TypeError(f"mygrad.save requires a Tensor-type object, got type {type(tensor)}")
+
+    # if tensor.grad == None:
+    #     np.savez(filename, data=tensor.data)
+    np.savez(filename, data=tensor.data, grad=tensor.grad)
+
+
+def load(tensor_filename : str) -> Tensor:
+    """Loads a saved Tensor and its gradient information (if applicable).
+
+    This docstring was adapted from that of numpy.load()
+
+    Parameters
+    ----------
+    tensor_filename : str
+        The name of the file that holds the tensor data to load.
+    
+    Returns
+    -------
+    A tensor with the desired gradient data.
+    """
+    _tensor = np.load(tensor_filename, allow_pickle=True)
+
+    loaded_tensor = tensor(_tensor['data'])
+
+    if not _tensor['grad'].all():
```

Comment:
This doesn't do quite what you want. ` _tensor['grad']` checks to see if the array consists of all non-zero values. So this actually returns `False` if the gradient happens to contain any zeros, which means that any such gradient will not be loaded by this current code.

Given that we will want `save` to only include a `grad` field in the `.npz` file, we can do the following:

```python
with np.load(file") as opened_file:
        loaded_tensor = tensor(opened_file["data"])
    
    if "grad" in opened_file:
        loaded_tensor.backward(f["grad"])
```

The `with np.load(file) as opened_file` line opens the `.npz` file using a [context manager](https://www.pythonlikeyoumeanit.com/Module5_OddsAndEnds/WorkingWithFiles.html#Opening-Files). Within the context, `file` acts kind of like a dictionary whose keys are the array-names you use to save the `.npz` file, and whose values are the associated arrays. Thus `"grad" in file` will return `True` only if you included a `grad` field in the saved file.

Note that by avoiding saving `grad` when `grad` is `None`, we can avoid needing to specify `allow_pickle=True` during loading. It turns out that `allow_pickle=True` can create security vulnerabilities in software, so it is nice that we can avoid this.